### PR TITLE
log aborted requests on hapi

### PR DESCRIPTION
**Pull request by jfromaniello**
Log aborted requests on the HAPI integration.

### CODE REVIEW

File: eventlogger/hapi_server.js [hunk 1]
```diff
@@ -8,6 +8,18 @@ module.exports.watch = function  (logger, server, options) {
 
   var ignorePaths = options.ignorePaths;
 
+  function getLogEntry(request, log_type) {
```
Comment (dschenkelman) [inline eventlogger/hapi_server.js]: can we rename to `createLogEntry`? `get` can make it look to the reader like it exists somewhere already

File: eventlogger/hapi_server.js [hunk 2]
```diff
@@ -18,22 +30,17 @@ module.exports.watch = function  (logger, server, options) {
         req: request
       });
     }
+
+    if (tags.request && tags.closed) {
+      const abortLogEntry = getLogEntry(request, 'request_aborted');
+      delete requests_start_time[request.id];
```
Comment (dschenkelman) [inline eventlogger/hapi_server.js]: this is already happening in line 13. I think we can remove line 36

File: eventlogger/hapi_server.js [hunk 3]
```diff
@@ -8,6 +8,18 @@ module.exports.watch = function  (logger, server, options) {
 
   var ignorePaths = options.ignorePaths;
 
+  function getLogEntry(request, log_type) {
+    const took = new Date() - requests_start_time[request.id];
```
Comment (elbuo8) [inline eventlogger/hapi_server.js]: `Date.now()`

File: eventlogger/hapi_server.js [hunk 4]
```diff
@@ -18,22 +30,17 @@ module.exports.watch = function  (logger, server, options) {
...
-      };
-      delete requests_start_time[request.id];
-      if (tags.closed && tags.error) {
-        log_event.err = new Error('connection closed');
-      }
-      if (tags.aborted && tags.error) {
-        log_event.err = new Error('connection aborted');
-      }
-      return logger.info(log_event);
+      const rsponseLogEntry = getLogEntry(request, 'response');
```
Comment (elbuo8) [inline eventlogger/hapi_server.js]: typo?

File: eventlogger/hapi_server.js [hunk 5]
```diff
@@ -8,32 +8,39 @@ module.exports.watch = function  (logger, server, options) {
...
-        log_type: 'response',
-        req:  request,
-        res:  request.response,
-        took: new Date() - requests_start_time[request.id]
-      };
-      delete requests_start_time[request.id];
-      if (tags.closed && tags.error) {
-        log_event.err = new Error('connection closed');
-      }
-      if (tags.aborted && tags.error) {
```
Comment (dschenkelman) [inline eventlogger/hapi_server.js]: where is `tags.aborted` handled in the new code?

**Pull request merged**